### PR TITLE
Uniform formating

### PR DESCRIPTION
**Pull request by Fritsch-Tech**
#### 🔧 Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [x] Codebase improvement
- [ ] Other (if none of the other choices apply)

#### 🚨 Checklist

- [x] I've read the [guidelines for contributing](https://github.com/gremo/react-directus/blob/main/CONTRIBUTING.md)
- [x] I've added necessary documentation (if appropriate)
- [x] I've ensured that my code additions do not fail linting or unit tests (if applicable)

#### Description

This pull request introduces several enhancements aimed at improving collaboration and maintaining a consistent codebase. The changes made include:

1. **Prettier Configuration File**: I've added a `.prettierrc` file to the project. This file includes the configuration settings for Prettier, a code formatter, to ensure that the code is uniformly formatted across the entire codebase.

2. **Pre-commit Hook**: I've implemented a pre-commit hook that automatically formats and lints the code before each commit. This will help prevent inconsistent code formatting from being committed, and ensure that the codebase remains clean and readable.

4. **Removed Conflicting ESLint Rules**: I have identified and removed conflicting ESLint rules from the project configuration. This resolves conflicts that could have caused issues with the codebase's linting process.

5. **Enforce Formatting During Linting**: With the updated ESLint configuration, the linting process now enforces code formatting rules as specified in Prettier. This ensures that the code consistently follows the defined formatting guidelines during development.

6. **Updated CONTRIBUTING.md**: I've made appropriate updates to the `CONTRIBUTING.md` file to include instructions on using the formating script.

##### Reason

Maintaining a standardized code format is crucial for a collaborative project. By introducing these changes, I aim to achieve the following:

1. **Better Collaboration**: With a consistent code format enforced by Prettier and the pre-commit hook, it will be easier for developers to review and understand each other's code, leading to smoother collaboration.

2. **Uniform Formatting**: Having a uniform code style throughout the project enhances code readability, making it easier for current and future contributors to dive into the codebase and contribute effectively.

3. **Easier Pull Request Review**: By using Prettier and the pre-commit hook to enforce consistent code formatting, the review process for pull requests becomes more straightforward. Reviewers can focus on the logic and functionality of the changes rather than spending time on code style issues. This streamlines the review process, leading to quicker and more effective code reviews, which benefits both contributors and reviewers alike.

### CODE REVIEW

File: .github/pull_request_template.md
Comment (gremo) [inline .github/pull_request_template.md:1]: I think that this file shouldn't be changed right?

File: .prettierrc.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "printWidth": 80,
```
Comment (gremo) [inline .prettierrc.json]: Maybe 120? I've always find 80 so limitating

File: package.json [hunk 2]
```diff
@@ -12,8 +12,13 @@
     "lint": "eslint src/* --ext .ts,.tsx",
     "test": "jest",
     "prepublishOnly": "npm run build",
-    "build": "rimraf dist/* && rollup -c --bundleConfigAsCjs"
+    "build": "rimraf dist/* && rollup -c --bundleConfigAsCjs",
+    "format": "npx prettier --write \"**/*.{js,jsx,ts,tsx,json,md}\""
```
Comment (gremo) [inline package.json]: Why npx prettier and not just prettier? Why also md files?

**Comment by gremo** [comment]
Huge work, thanks. I've added some comments. I'm OK with using prettier!

File: .github/pull_request_template.md
Comment (Fritsch-Tech) [reply .github/pull_request_template.md:1]: Per github styling convention `* *` and `_ _` are equivalent.
[Text Styling](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#styling-text)

Comment (gremo) [reply .github/pull_request_template.md:1]: I know, I prefer not to prettier .md files too :)

File: .prettierrc.json (hunk 1 above)
Comment (Fritsch-Tech) [reply .prettierrc.json]: Seams good to me, especially with the current wide screen monitors.
I will make a new commit.

File: package.json (hunk 2 above)
Comment (Fritsch-Tech) [reply package.json]: I originally didn't add Prettier as a dev dependency, but now I can remove the npx.
I generally prefer to format the Markdown files as well, but that is a personal preference.

Comment (gremo) [reply package.json]: Well, let's exclude Markdown as it's already markdownlint for that. Formatting is not crucial, I'm usually very precise and tidy
 about that 😄

Comment (Fritsch-Tech) [reply package.json]: I fully excluded it in the new commit and reverted the changed Markdown files back to there original state.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -15,10 +15,10 @@ In order to contribute to this project, follow a few easy steps:
...
-4. Run `npm run lint`, `npm run test` and `npm run build` and verify that they complete without errors
-5. Push `my-awesome-feature` branch to GitHub and open a [pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests)
+4. Run `npm run format`, `npm run lint`, `npm run test` and `npm run build`,  
+   or `npm run format && npm run lint && npm run test && npm run build` and verify that they complete without errors5. Push `my-awesome-feature` branch to GitHub and open a [pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests)
 6. Liked some of my work? Buy me a ☕ (or more likely 🍺)
 
 <p align="center">
   <a href="https://paypal.me/marcopolichetti" target="_blank"><img src="https://img.shields.io/badge/Donate-PayPal-ff3f59.svg"/></a>
-</p>
+</p>
```
Comment (gremo) [inline CONTRIBUTING.md]: What's the difference here? Maybe LF?

File: README.md [hunk 4]
```diff
@@ -140,4 +140,4 @@ All types of contributions are encouraged and valued. See the [Contributing](CON
 
 ## 📘 License
 
-This project is released under the under terms of the [ISC License](LICENSE).
+This project is released under the under terms of the [ISC License](LICENSE).
```
Comment (gremo) [inline README.md]: What's the difference here? Maybe LF?

File: package.json [hunk 5]
```diff
@@ -12,8 +12,13 @@
     "lint": "eslint src/* --ext .ts,.tsx",
     "test": "jest",
     "prepublishOnly": "npm run build",
-    "build": "rimraf dist/* && rollup -c --bundleConfigAsCjs"
+    "build": "rimraf dist/* && rollup -c --bundleConfigAsCjs",
+    "format": "prettier --write \"**/*.{js,jsx,ts,tsx,json}\""
```
Comment (gremo) [inline package.json:16]: If using prettierignore, why not simplify this to "prettier --write"?

Comment (Fritsch-Tech) [reply package.json:16]: You are right, that way we only have a single source of truth for excluding files.

File: CONTRIBUTING.md (hunk 3 above)
Comment (Fritsch-Tech) [reply CONTRIBUTING.md]: Yes, i managed to reverte it by overwriting it with the original source.

File: README.md (hunk 4 above)
Comment (Fritsch-Tech) [reply README.md]: Yes, i managed to reverte it by overwriting it with the original source.

File: CONTRIBUTING.md (hunk 3 above)
Comment (gremo) [reply CONTRIBUTING.md]: Ok then fix, squash all commits and I'll merge! 🥳

Comment (Fritsch-Tech) [reply CONTRIBUTING.md]: Perfect, i squashed it all now

Comment (gremo) [reply CONTRIBUTING.md]: Thanks, it seems all fine except line 24 of contributing... still LF?

Comment (Fritsch-Tech) [reply CONTRIBUTING.md]: You are right, sory for the inconvinience.
I fixed it now.

**Comment by gremo** [comment]
Awesome job thanks!

**Pull request merged**